core/src/main/java/org/apache/druid/java/util/http/client/response/FullResponseHolder.java

```diff
@@ -22,23 +22,15 @@
 import org.jboss.netty.handler.codec.http.HttpResponse;
 import org.jboss.netty.handler.codec.http.HttpResponseStatus;
 
-/**
- */
-public class FullResponseHolder
+public abstract class FullResponseHolder<T>
```

Comment:
If a `ReponseHolder` interface was added, would that simplify any of the changes? I think if that had already been in place, many of the test modification you made may have not been needed, for example.

Reply:
Hmm, I'm not sure if it helps. I think the most of the changes in unit tests are just rename since I changed `FullResponseHolder.getContent()` to `FullResponseHolder.getAccumulated()`.

Reply:
Yeah, those would have to change because the method name changed. However, I was thinking that a lot of the changes from `FullResponseHandler` to `StringFullResponseHandler` would not have been needed if the code had been referencing an interface instead of an implementation.

Reply:
Those test are for `KafkaIndexTaskClient`, `KinesisIndexTaskClient`, and `RemoteTaskActionClient` which use `StringFullResponseHandler` for type safety. I think it won't be easy to mock `FullResponseHandler` in those tests.